tests/python/contrib/test_hexagon/topi/test_conv2d_transpose.py

```diff
@@ -0,0 +1,177 @@
+    return num_filter
+
+
+@tvm.testing.fixture
+def shift_shape(stride):
+    return stride
+
+
+@tvm.testing.fixture
+def shift_shape(padding):
+    return padding
+
+
+@tvm.testing.fixture
+def shift_shape(output_padding):
+    return output_padding
+
+
+class BaseConv2DTransposeTests:
+    @requires_hexagon_toolchain
+    def test_conv2d(
+        self,
+        hexagon_session,
+        batch,
+        in_channel,
+        in_size,
+        num_filter,
+        stride,
+        padding,
+        output_padding,
+        random_seed,
+    ):
+
+        target_hexagon = tvm.target.hexagon("v68")
+
+        in_height, in_width = in_size
+        kernel_height, kernel_width = (1, 1)
+        stride_height, stride_width = stride
+        pad_top, pad_left, pad_bottom, pad_right = padding
+
+        A = te.placeholder((batch, in_channel, in_height, in_width), name="A")
+        W = te.placeholder((in_channel, num_filter, kernel_height, kernel_width), name="W")
+
+        a_shape = get_const_tuple(A.shape)
+        w_shape = get_const_tuple(W.shape)
+        dtype = A.dtype
+
+        def get_ref_data():
+
+            np.random.seed(random_seed)
+            a_np = np.random.uniform(size=a_shape).astype(dtype)
+            w_np = np.random.uniform(size=w_shape).astype(dtype)
+            b_np = tvm.topi.testing.conv2d_transpose_nchw_python(
+                a_np, w_np, stride, padding, output_padding
+            )
+            c_np = np.maximum(b_np, 0)
+            return a_np, w_np, b_np, c_np
+
+        a_np, w_np, b_np, c_np = get_ref_data()
+
+        fcompute_args = (
+            A,
+            W,
+            [stride_height, stride_width],
+            [pad_top, pad_left, pad_bottom, pad_right],
+            A.dtype,
+            output_padding,
+        )
+
+        with tvm.target.Target(target_hexagon):
+            fcompute = topi.nn.conv2d_transpose_nchw
+            fschedule = topi.hexagon.schedule_conv2d_transpose_nchw
+            B = fcompute(*fcompute_args)
+            C = topi.nn.relu(B)
+            s1 = fschedule([B])
+            s2 = fschedule([C])
+
+            dev = hexagon_session.device
+
+            a = tvm.nd.array(a_np, dev)
+            w = tvm.nd.array(w_np, dev)
+            b = tvm.nd.array(np.zeros(get_const_tuple(B.shape), dtype=B.dtype), dev)
+            c = tvm.nd.array(np.zeros(get_const_tuple(C.shape), dtype=C.dtype), dev)
+
+            func1 = tvm.build(s1, [A, W, B], tvm.target.Target(target_hexagon, host=target_hexagon))
+            func2 = tvm.build(s2, [A, W, C], tvm.target.Target(target_hexagon, host=target_hexagon))
+
+            mod1 = hexagon_session.load_module(func1)
+            mod2 = hexagon_session.load_module(func2)
+
+            mod1(a, w, b)
+            mod2(a, w, c)
+            tvm.testing.assert_allclose(b.numpy(), b_np, rtol=1e-5)
+            tvm.testing.assert_allclose(c.numpy(), c_np, rtol=1e-5)
+
+
+class TestConv2DTranspose_1(BaseConv2DTransposeTests):
```

Comment:
can you combine these three classes using this patter:
batch, in_channel, ... = tvm.testing.parameters((1, (3, 8), ...)

Reply:
Thanks for the suggestion. Have addressed both in the next commit.